instance/tests/models/test_openedx_site_configuration_mixin.py

```diff
@@ -128,3 +128,109 @@ def test_html_elements_and_attributes_in_static_content_overrides(self):
             ]
         }
         self.assertEqual(yaml.safe_load(instance.get_site_configuration_settings()), expected_variables)
+
+    def test_app_server_site_configuration_override(self):
+        """
+        Test that when the static content overrides contain html elements and attributes, there are no errors and the
+        expected ansible SiteConfiguration variables are generated.
+        """
+        instance = OpenEdXInstanceFactory()
+        configuration_extra_settings = {
+            'EDXAPP_SITE_CONFIGURATION': [
+                {
+                    'values': {
+                        'override': True,
+                        'CONTACT_US_CUSTOM_LINK': '/random/place/'
+                    }
+                }
+            ]
+        }
+        instance.configuration_extra_settings = yaml.dump(
+            configuration_extra_settings,
+            default_flow_style=False,
+        )
+
+        static_content_overrides = {
+            'version': 0,
+            'static_template_about_content': 'static_template_about_content',
+            'homepage_overlay_html': 'homepage_overlay_html',
+        }
+        instance.static_content_overrides = static_content_overrides
+        app_server = instance._create_owned_appserver()
+
+        expected_variables = [
+            {
+                'values': {
+                    'override': True,
+                    'CONTACT_US_CUSTOM_LINK': '/random/place/',
+                    'static_template_about_content': 'static_template_about_content',
+                    'homepage_overlay_html': 'homepage_overlay_html',
+                }
+            }
+        ]
+
+        configuration_settings = yaml.load(app_server.configuration_settings, Loader=yaml.SafeLoader)
+        self.assertEqual(configuration_settings['EDXAPP_SITE_CONFIGURATION'], expected_variables)
+
+    def test_app_server_site_configuration_override_complex(self):
+        """
+        Test that when the static content overrides contain html elements and attributes, there are no errors and the
+        expected ansible SiteConfiguration variables are generated.
+        """
+        instance = OpenEdXInstanceFactory()
+        instance.configuration_extra_settings = yaml.dump({
+            "EDXAPP_SITE_CONFIGURATION": [
+                {
+                    "values": {
+                        "SOME_SETTING": "some value"
+                    }
+                },
+                {
+                    "domain": "some-specific-site.com",
+                    "values": {
+                        "SPECIFIC_SETTING": "specific value"
+                    }
```

Comment:
Small nit: Trailing comma please.

```suggestion
                    },
```